ts/components/CompositionArea.tsx

```diff
@@ -357,6 +357,37 @@ export const CompositionArea = ({
     };
   }, [setLarge]);
 
+  // Listen to any key in the conversation panel to
+  React.useEffect(() => {
+    const handler = (e: KeyboardEvent) => {
+      const { key } = e;
+
+      // We don't want to react to a control character
+      if (key.length !== 1) {
+        return;
+      }
+
+      // We don't want to switch focus if another panel is up
+      const panels = document.querySelectorAll('.conversation .panel');
+      if (panels && panels.length > 1) {
+        return;
+      }
+
+      // We don't want to take focus away of input fields
+      const { activeElement } = document;
+      if (activeElement?.nodeName.toLowerCase() === 'input') {
+        return;
+      }
+
+      inputApiRef.current?.focus();
```

Comment:
Should this use [the `focusInput` callback](https://github.com/signalapp/Signal-Desktop/blob/6700d2f196e674b2e0778f56cb18e38f862a9c39/ts/components/CompositionArea.tsx#L327-L329) instead?

If yes, then:

```suggestion
      focusInput();
```